WordPress/gutenberg
# Grid Visualizer - improve styles

## What?
See #64162

Brings the styles of the grid visualizer closer to those shown in the video on the issue.

There are a few bits this doesn't do from the video:
- The grid isn't revealed on hover, I feel this is better to do in a separate PR
- Dragging to select multiple grid cells is also out of scope for this PR
- In auto mode, the video shows the next available cell being interactive like the manual mode cells. This should be possible in a follow up. 

There are a few other subtleties that might worth trying in separate PRs:
- There's quite a lot of visual noise if you have a block with the visualizer squares under it. We could try de-emphasizing the cells under the selected child block within a grid or not showing them at all. The video possibly shows something like this at 1:55, but hard to say for sure

## How?
In order to achieve the visual styling shown in #64162, the grid visualizer needs to be rendered behind the grid block.

This PR attempts this using an inline popover.

It requires quite a bit of code-reshuffling as there are now two separate grid visualizers being rendered:
- `GridPopunder` - renders a visualizer under the selected grid block. This visualizer is entirely visual, it renders some blue rectangles.
- `GridPopover` - renders interactive elements over the selected block. This is responsible for the drop zones and appenders in the grid. It has to be over the selected block to receive mouse events.

## Testing Instructions
1. Add a grid block
2. Switch between auto and manual and notice the cells are blue in manual mode to indicate interactivity
3. In manual mode, try dragging a block into an empty cell and observe the highlight styles.
4. Hover over an empty cell and observe the appender styles

## Screenshots or screencast 
![Screenshot 2024-08-09 at 5 56 41 PM](https://github.com/user-attachments/assets/231f3fa7-2f98-4806-a7dc-050209537e31)
